sway-core/src/transform/to_parsed_lang/convert_parse_tree.rs

```diff
@@ -2044,6 +2045,199 @@ fn expr_func_app_to_expression_kind(
                 },
             ));
         }
+        // "__dbg(arg)" in debug becomes "{
+        //      let mut f = Formatter { };
+        //      f.print_str("[{current_file}:{current_line}:{current_col}] = ");
+        //      let arg = arg;
+        //      arg.fmt(f);
+        //      arg
+        // }"
+        Some(Intrinsic::Dbg)
+            if context.is_dbg_generation_full() && last.is_none() && !is_relative_to_root =>
+        {
+            if arguments.len() > 1 {
```

Comment:
The check should actually be `arguments.len() != 1`. With this check, if we have only `__dbg()` call without arguments the compiler will panic below in the `arguments[0].clone()` call.

An interesting question is if we actually want to support empty `__dbg()` call like in Rust that will print only the file, line, and column information? :thinking: What do you think about that?

Reply:
> An interesting question is if we actually want to support empty __dbg() call like in Rust that will print only the file, line, and column information? 🤔 What do you think about that?